src/viam/ur/module/ur_arm.cpp

```diff
@@ -215,6 +215,20 @@ URArm::URArm(Model model, const Dependencies& deps, const ResourceConfig& cfg) :
         throw std::runtime_error("couldn't connect to dashboard");
     }
 
+    // verify that the connected arm is the same model as the configured module
+    {
+        std::string conn_model_type{};
+        if (!current_state_->dashboard->commandGetRobotModel(conn_model_type)) {
+            throw std::runtime_error("failed to get model info of connected arm");
+        }
+
+        std::string cfg_model_type = model_type();
```

Comment:
- const, but see below
- I'd call this `configured_model_type` - no need to shorten it. Autocomplete means you never need to type it anyway.